File: Swift/基本语法/第二部分/06-block的回顾/06-block的回顾/HttpTool.m
//
//  HttpTool.m
//  06-block的回顾
//
//  Created by tzsoft on 2018/2/23.
//  Copyright © 2018年 胡小羊. All rights reserved.
//

#import "HttpTool.h"

@implementation HttpTool
- (void)loadData:(void (^)(NSString *))callBack{
    //
    dispatch_async(dispatch_get_global_queue(0, 0),^{
        // 回调主线程
        dispatch_sync(dispatch_get_main_queue(),^{
            callBack(@"哈哈");
        });
    });
}
@end

File: Swift/基本语法/第二部分/08-尾随闭包/07-闭包的使用/HttpTool.swift
//
//  HttpTool.swift
//  07-闭包的使用
//
//  Created by tzsoft on 2018/2/26.
//  Copyright © 2018年 胡小羊. All rights reserved.
//

import UIKit

class HttpTool: NSObject {

    var callBack:((String ,String)->() )?
    
    // 闭包的类型：(参数类型) -> (返回值类型)
    func loadData(callBack: @escaping (String,String)->()) {
        print("发送网络请求")
        self.callBack  = callBack
        callBack("数据这是", "1231312")
    }
}
